cups/array.c

```diff
@@ -787,9 +787,9 @@ _cupsArrayNewStrings(const char *s,	/* I - Delimited strings or NULL */
   cups_array_t	*a;			/* Array */
 
 
-  if ((a = cupsArrayNew3((cups_array_func_t)strcmp, NULL, NULL, 0,
-                         (cups_acopy_func_t)_cupsStrAlloc,
-			 (cups_afree_func_t)_cupsStrFree)) != NULL)
+  if ((a = cupsArrayNew3((cups_array_func_t)_cupsArrayStrcmp, NULL, NULL, 0,
+                         (cups_acopy_func_t)_cupsStrPrivAlloc,
```

Comment:
Let's use your `_cupsArrayStrdup` and `_cupsArrayFree` here, no need to use the string pool functions...